### PR TITLE
`@schema/desc` adds `description` key to OpenAPI Doc

### PR DESCRIPTION
**Pull request by cari-lynn**
When a schema node is annotated with `@schema/desc` and exported as an openapi document, the `description` key is added.

### CODE REVIEW

File: pkg/schema/annotations.go [hunk 1]
```diff
@@ -146,6 +192,11 @@ func (n *DefaultAnnotation) NewTypeFromAnn() yamlmeta.Type {
 	return nil
 }
 
+// NewTypeFromAnn returns type information given by annotation. DescriptionAnnotation has no type information.
+func (n *DescriptionAnnotation) NewTypeFromAnn() yamlmeta.Type {
+	return nil
+}
+
```
Comment (pivotaljohn) [inline pkg/schema/annotations.go]: Can we massage `Annotation` into something more broadly suitable?... 🤔 

Here's an attempt:

I think we're "discovering" three (3) types of annotations:
- `TypingAnnotation` — contributes to the schema-type of the node
- `ValueHavingAnnotation` — contributes to the value of the node
- `DocumentingAnnotation` — contributes to the description of the node/type

And we'd break up `collectionAnnotation()` into three separate (very similar) functions that gather each collection.

File: pkg/schema/openapi.go [hunk 2]
```diff
@@ -74,16 +86,26 @@ func (o *OpenAPIDocument) calculateProperties(schemaVal interface{}) *yamlmeta.M
...
 	case *AnyType:
-		return &yamlmeta.Map{Items: []*yamlmeta.MapItem{
+		m := &yamlmeta.Map{Items: []*yamlmeta.MapItem{
 			{Key: "nullable", Value: true},
 			{Key: "default", Value: typedValue.GetDefaultValue()},
 		}}
+		if typedValue.GetDescription() != "" {
+			m.Items = append(m.Items, &yamlmeta.MapItem{Key: "description", Value: typedValue.GetDescription()})
+		}
+		return m
```
Comment (pivotaljohn) [inline pkg/schema/openapi.go]: We're using two different names for the same variable: `properties` vs. `m`. At this point, it's a difference without a reason (at least that I can see, right now).

Let's make them all consistent.

File: pkg/schema/type.go [hunk 3]
```diff
@@ -17,6 +17,7 @@ var _ yamlmeta.Type = (*ArrayType)(nil)
 var _ yamlmeta.Type = (*ArrayItemType)(nil)
 var _ yamlmeta.Type = (*AnyType)(nil)
 var _ yamlmeta.Type = (*NullType)(nil)
+var _ yamlmeta.Type = (*ScalarType)(nil)
```
Comment (pivotaljohn) [inline pkg/schema/type.go:20]: Nice catch; thank you.

File: pkg/schema/type.go [hunk 4]
```diff
@@ -96,8 +102,8 @@ func (a *ArrayItemType) SetDefaultValue(val interface{}) {
 }
 
 // SetDefaultValue sets the default value to `val`
-func (a *ScalarType) SetDefaultValue(val interface{}) {
-	a.defaultValue = val
+func (m *ScalarType) SetDefaultValue(val interface{}) {
+	m.defaultValue = val
```
Comment (pivotaljohn) [inline pkg/schema/type.go]: I can see `s` or `t` ... but `m`? 😁

File: pkg/schema/openapi.go [hunk 5]
```diff
@@ -52,17 +52,29 @@ func (o *OpenAPIDocument) calculateProperties(schemaVal interface{}) *yamlmeta.M
 		property := yamlmeta.Map{Items: []*yamlmeta.MapItem{
 			{Key: "type", Value: "object"},
 			{Key: "additionalProperties", Value: false},
-			{Key: "properties", Value: &yamlmeta.Map{Items: properties}},
 		}}
+
+		if typedValue.GetDescription() != "" {
+			property.Items = append(property.Items, &yamlmeta.MapItem{Key: "description", Value: typedValue.GetDescription()})
```
Comment (pivotaljohn) [inline pkg/schema/openapi.go]: Given that we're using the same string in multiple places, should we extract the OpenAPI property names into constants?

**Review by pivotaljohn — APPROVED** [review]
Overall, LGTM. 👍🏻 

Comments are reacting to some internal structure thoughts... these can easily be addressed in the upcoming stories too.

File: pkg/schema/annotations.go (hunk 1 above)
Comment (cari-lynn) [reply pkg/schema/annotations.go]: Great suggestion, added a function for this

File: pkg/schema/openapi.go [hunk 6]
```diff
@@ -9,6 +9,14 @@ import (
 	"github.com/k14s/ytt/pkg/yamlmeta"
 )
 
+// keys used when generating an OpenAPI Document
+const (
+	Type        = "type"
+	Default     = "default"
+	Nullable    = "nullable"
+	Description = "description"
```
Comment (pivotaljohn) [inline pkg/schema/openapi.go]:
```suggestion
	typeProp        = "type"
	defaultProp     = "default"
	nullableProp    = "nullable"
	descriptionProp = "description"
```

Suggestions:
- un-export these constants as they are not intended to be used outside the package.
- need to disambiguate `type` from the Go keyword... for example add the `Prop` suffix. 🤷🏻

**Review by pivotaljohn — APPROVED** [review]
One request to un-export the constants for the OpenAPI property names, but otherwise GTG.

(feel free to merge after addressing that suggestion)

**Pull request merged**